easybuild/easyblocks/p/python.py

```diff
@@ -141,15 +143,25 @@ def prepare_for_extensions(self):
 
     def configure_step(self):
         """Set extra configure options."""
-        self.cfg.update('configopts', "--with-threads --enable-shared")
+        self.cfg.update('configopts', "--enable-shared")
 
+        # Explicitely enable thread support on < 3.7 (always on 3.7+)
+        if LooseVersion(self.version) < LooseVersion('3.7'):
+            self.cfg.update('configopts', "--with-threads")
+
+        # Explicitely enable unicode on Python 2, always on for Python 3
         # Need to be careful to match the unicode settings to the underlying python
-        if sys.maxunicode == 1114111:
-            self.cfg.update('configopts', "--enable-unicode=ucs4")
-        elif sys.maxunicode == 65535:
-            self.cfg.update('configopts', "--enable-unicode=ucs2")
-        else:
-            raise EasyBuildError("Unknown maxunicode value for your python: %d" % sys.maxunicode)
+        if LooseVersion(self.version) < LooseVersion('3.0'):
+            if sys.maxunicode == 1114111:
+                self.cfg.update('configopts', "--enable-unicode=ucs4")
+            elif sys.maxunicode == 65535:
+                self.cfg.update('configopts', "--enable-unicode=ucs2")
+            else:
+                raise EasyBuildError("Unknown maxunicode value for your python: %d" % sys.maxunicode)
+
+        # Enable further optimizations at the cost of a longer build
```

Comment:
Maybe clarify the comment to explain the version check >= 3.5.3?

Bonus points for a reference to the relevant section in the Python documentation...

Reply:
https://docs.python.org/3.7/whatsnew/changelog.html#python-3-5-3-release-candidate-1

> bpo-26359: Rename –with-optimiations to –enable-optimizations.

This seems to be where it was added initially, see the linked issue in the changelog. I also downloaded a couple python releases (source) and grepped for this option and it seems to consistently appear after 3.5.3

Reply:
OK, just clarify that a bit in the comment then?

Reply:
ok